core/models.py

```diff
@@ -0,0 +1,125 @@
+import hashlib
+import re
+from typing import ClassVar
+
+from django.contrib.auth import get_user_model
+from django.contrib.contenttypes.fields import GenericForeignKey, GenericRelation
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+User = get_user_model()
+
+
+class GenericModel(models.Model):
+    user = models.ForeignKey(User, on_delete=models.SET_NULL, null=True)
+    is_active = models.BooleanField(default=True)
+    is_locked = models.BooleanField(default=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:  # noqa: D106
+        abstract = True
+
+
+class Community(GenericModel):
+    name = models.CharField(max_length=255)
+
+    def __str__(self) -> str:  # noqa: ANN101, D105

```

Comment:
Fix please. 
ANN101: https://docs.astral.sh/ruff/rules/missing-type-self/
D105 - add to ingores in ruff.